src/compiler/scanner.ts

```diff
@@ -1066,15 +1122,8 @@ export function createScanner(languageVersion: ScriptTarget, skipTrivia: boolean
 
     return scanner;
 
-    function error(message: DiagnosticMessage): void;
-    function error(message: DiagnosticMessage, errPos: number, length: number, arg0?: any): void;
-    function error(message: DiagnosticMessage, errPos: number = pos, length?: number, arg0?: any): void {
-        if (onError) {
-            const oldPos = pos;
-            pos = errPos;
-            onError(message, length || 0, arg0);
-            pos = oldPos;
-        }
+    function error(message: DiagnosticMessage, errPos = pos, length = 0, ...args: DiagnosticArguments): DiagnosticWithDetachedLocation | undefined {
+        return onError?.(message, errPos, length, ...args);
     }

```

Comment:
I wonder if this refactor would cause serious compatibility issue, as what the TypeScript Bot warns. Nevertheless provided that an extra argument makes the code neat and tidy, setting the variable back and forth sounds ridiculous to me.